Use issue form for bug reports

trevorbayless opened the pull request:
Removes the existing markup bug template and replaces it with an [issue form](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms). I've kept all of the previous template fields for consistency. The benefits of using issue forms is that it allows for fields to be specified as required. This will allow for better bug reporting as reporters cannot submit a bug unless the required fields are filled out.

I've setup an [example repo with this exact issue form](https://github.com/trevorbayless/test-github-issue-form/issues/new?assignees=&labels=bug&template=bug.yml) for demo purposes in order to see how it will be rendered.

There shouldn't be any changes needed on the repo side (such as enabling the template) to make this work. It should be enabled by default after merging.

benediktwerner commented:
Looks good.

Can we link `please search to confirm this is not a duplicate issue` to the issues page? Maybe `please [search existing issues](https://github.com/lichess-org/lila/issues) to confirm ...`.

Also might make sense to use `e.g. Windows 11` for the OS and browser placeholders? Looks slightly confusing currently (as if it's prefilled).

And I guess there are bugs (e.g. API-related) where it doesn't really make sense to specify the browser 🤔 But maybe still a good idea to keep it required since it's frequently not provided when it would be crucial information.

trevorbayless commented:
Thanks for your feedback. I've linked the issues page and also removed the placeholders. I added a description to the "Browser and Version" section to mention specifying API if that's your access method. I've updated the demo URL if you want to check it out again.
